api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/WorkspaceTargetProperty.java

```diff
@@ -0,0 +1,80 @@
+package org.pmiops.workbench.actionaudit.targetproperties;
+
+import java.util.function.Function;
+import org.jetbrains.annotations.NotNull;
+import org.pmiops.workbench.model.ResearchPurpose;
+import org.pmiops.workbench.model.SpecificPopulationEnum;
+import org.pmiops.workbench.model.Workspace;
+
+// N.B. entries will rarely be referred to by name, but are accessed via values(). I.e. they are
+// not safe to remove, even if an IDE indicates otherwise.
+public enum WorkspaceTargetProperty implements ModelBackedTargetProperty<Workspace> {
+  ETAG("etag", Workspace::getEtag),
+  NAME("name", Workspace::getName),
+  NAMESPACE("namespace", Workspace::getNamespace),
+  CDR_VERSION_ID("cdr_version_id", Workspace::getCdrVersionId),
+  CREATOR("creator", Workspace::getCreator),
+  ACCESS_TIER_SHORT_NAME("access_tier_short_name", Workspace::getAccessTierShortName),
+  ADDITIONAL_NOTES(
+      "additional_notes", workspace -> workspace.getResearchPurpose().getAdditionalNotes()),
+  APPROVED("approved", rpStringOrNull(ResearchPurpose::isApproved)),
+  ANCESTRY("ancestry", rpStringOrNull(ResearchPurpose::isAncestry)),
+  ANTICIPATED_FINDINGS(
+      "anticipated_findings", workspace -> workspace.getResearchPurpose().getAnticipatedFindings()),
+  COMMERCIAL_PURPOSE("commercial_purpose", rpStringOrNull(ResearchPurpose::isCommercialPurpose)),
+  CONTROL_SET("control_set", rpStringOrNull(ResearchPurpose::isControlSet)),
+  DISEASE_FOCUSED_RESEARCH(
+      "disease_focused_research", rpStringOrNull(ResearchPurpose::isDiseaseFocusedResearch)),
+  DISEASE_OF_FOCUS(
+      "disease_of_focus", workspace -> workspace.getResearchPurpose().getDiseaseOfFocus()),
+  DRUG_DEVELOPMENT("drug_development", rpStringOrNull(ResearchPurpose::isDrugDevelopment)),
+  EDUCATIONAL("educational", rpStringOrNull(ResearchPurpose::isEducational)),
+  INTENDED_STUDY("intended_study", workspace -> workspace.getResearchPurpose().getIntendedStudy()),
+  METHODS_DEVELOPMENT("methods_development", rpStringOrNull(ResearchPurpose::isMethodsDevelopment)),
+  OTHER_POPULATION_DETAILS(
+      "other_population_details",
+      workspace -> workspace.getResearchPurpose().getOtherPopulationDetails()),
+  POPULATION_DETAILS(
+      "population_details",
+      workspace -> {
+        var populationDetails = workspace.getResearchPurpose().getPopulationDetails();
+        if (populationDetails == null) return null;
```

Comment:
matches original Kotlin: if the field is missing entirely, return null.  If it's present but an empty list, return **empty string** instead.